src/operator/catch.ts

```diff
@@ -64,6 +65,8 @@ import { subscribeToResult } from '../util/subscribeToResult';
  * @name catch
  * @owner Observable
  */
+export function _catch<T>(this: Observable<T>, selector: (err: any, caught: Observable<T>) => ErrorObservable): Observable<T>;
+export function _catch<T, R>(this: Observable<T>, selector: (err: any, caught: Observable<T>) => ObservableInput<R>): Observable<T | R>;
```

Comment:
I guess I don't understand enough about the typescript compiler. Why do you need to duplicate this signature? Since it is directly below?